x-pack/platform/plugins/shared/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -112,6 +118,54 @@ export class KnowledgeBaseService {
     }));
   }
 
+  private async recallFromIntegrationsKnowledge({
+    queries,
+    esClient,
+  }: {
+    queries: Array<{ text: string; boost?: number }>;
+    esClient: { asCurrentUser: ElasticsearchClient; asInternalUser: ElasticsearchClient };
+  }): Promise<RecalledEntry[]> {
+    try {
+      const indexExists = await esClient.asInternalUser.indices.exists({
```

Comment:
while working on the security AI assistant piece for this work (https://github.com/elastic/kibana/pull/236197), I found that `kibana_system` does not have permission to call `indices.exists()`, nor does it have permission to call `indices.get()`. both calls would result in an error being thrown from ES

yes... it's weird, apparently `read` permissions for this index is apparently not sufficient to access these two "simple" index checks

we decided to implement an empty search as a workaround to check against index existence:
https://github.com/elastic/kibana/blob/9732c29117c05cfc471ceb1c0f111664d6a65e00/x-pack/solutions/security/plugins/security_solution/server/assistant/tools/integration_knowledge/integration_knowledge_tool.ts#L50-L57